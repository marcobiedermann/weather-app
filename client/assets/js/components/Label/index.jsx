import PropTypes from 'prop-types';
import React from 'react';

const Label = (props) => {
  const { children, htmlFor } = props;

  return (
    <label htmlFor={htmlFor}>
      {children}
    </label>
  );
};

Label.propTypes = {
  children: PropTypes.node,
  htmlFor: PropTypes.string,
};

Label.defaultProps = {
  children: null,
  htmlFor: '',
};

export default Label;
