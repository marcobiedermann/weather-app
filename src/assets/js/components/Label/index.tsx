/* eslint-disable jsx-a11y/label-has-associated-control */

import React, { FC } from 'react';

export interface LabelProps {
  htmlFor: string;
}

const Label: FC<LabelProps> = (props) => {
  return <label {...props} />;
};

export default Label;
