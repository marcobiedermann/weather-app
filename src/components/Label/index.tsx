import type { ReactNode } from 'react';

interface LabelProps {
  children: ReactNode;
  htmlFor: string;
}

function Label(props: LabelProps): JSX.Element {
  return <label {...props} />;
}

export type { LabelProps };
export default Label;
