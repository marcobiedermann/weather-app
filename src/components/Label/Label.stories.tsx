import type { Meta, StoryObj } from '@storybook/react';
import Label from '.';

const meta: Meta<typeof Label> = {
  args: {
    children: 'Label',
    htmlFor: 'username',
  },
  component: Label,
};

type Story = StoryObj<typeof Label>;

const Default: Story = {};

export { Default };
export default meta;
