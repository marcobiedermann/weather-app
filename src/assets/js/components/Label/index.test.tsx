import React from 'react';
import renderer from 'react-test-renderer';
import Label from '.';

describe('components/Label', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const tree = renderer.create(<Label htmlFor="foo">Content</Label>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
