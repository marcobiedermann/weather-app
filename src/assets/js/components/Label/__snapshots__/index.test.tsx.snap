// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Label renders correctly 1`] = `
<DocumentFragment>
  <label
    for="foo"
  >
    Content
  </label>
</DocumentFragment>
`;
